config/config-api/src/com/thoughtworks/go/domain/UnitTestReportGenerator.java

```diff
@@ -154,8 +156,8 @@ private void pumpFileContent(File file, PrintStream out) throws IOException {
         BufferedReader bufferedReader = null;
         try {
             bufferedReader = new BufferedReader(new FileReader(file));
-            String line = bufferedReader.readLine();
-            if (!line.contains("<?xml")) { // skip prolog
+            String line = PROLOG_PATTERN.matcher(bufferedReader.readLine()).replaceFirst("");
+            if (!line.isEmpty()) { // skip if line was only prolog
```

Comment:
I agree, it looks like this could go in `XmlUtils` and be tested in isolation there as well.